start r5 branch to prepare the release notes (#9)
SovereignCloudStack/release-notes: Release5.md

* The services `minio.services.osism.tech` and `harbor.services.osism.tech` are deprecated and will be turned of on October 20th, 2023.

## Deprecations

### Deprecations via OSISM

For these please also refer to the [upstream deprecation notices](https://release.osism.tech/notes/6.0.0.html#deprecations).

* It is again noted that the old scripts of the form ``osism-`` will be removed in the future.
  A note has been added to the scripts showing this when they are executed.

* The following services are deprecated and will be removed with R6 (OSISM 6.1.0):
  * Patchman
  * Adminer
  * Patchman Client
  * Virtualbmc
  * Bird

## Security Fixes

Throughout the Release 5 development cycle, the SCS project issued two security advisories for upstream components:

* In April 2023 an advisory in Open vSwitch (OvS) ([CVE-2023-1668](https://cve.report/CVE-2023-1668) was issued.
Our [advisory](https://scs.community/security/2023/04/21/cve-2023-1668/).

* In May 2023 an advisory affecting the OpenStack component Cinder ([CVE-2023-2088](https://cve.report/CVE-2023-2088)) was issued.
Our [advisory](https://scs.community/security/2023/05/10/cve-2023-2088/).

## Resolved Issues

> Numerous minor issue have been resolved. The most important steps on the IaaS side probably being the move to ceph Quincy

Reviewer: ```suggestion
Numerous minor issue have been resolved. The most important steps on the IaaS side probably being the move to ceph Quincy.
```

Reviewer 2: No, the sentence continues after the line break.

Reviewer: > No, the sentence continues after the line break.

sorry, my bad.

* [Cloud-in-a-Box](https://github.com/osism/cloud-in-a-box) now comes with Swift enabled as well as the option
for secondary NIC for external connectivity.

### Container Management

* The Kubernetes Cluster Management solution is [available as version 6.0.0](https://github.com/SovereignCloudStack/k8s-cluster-api-provider/blob/main/Release-Notes-R5.md)
* Kubernetes v1.24 .. 1.27 are officially supported. v1.28 also works (technical preview until officially supported by capo) as do older versions (with downgrading nginx-ingress), matching OCCM and CSI versions.
* Cluster-API (capi) v1.5.1, Cluster-API provider for Openstack (capo) v0.7.3 
* The node images now use Ubuntu 22.04, the management host can use Ubuntu 22.04 or Debian 12.
* Cilium v1.14.1, default now, though Calico (3.26.x) is still supported.
* Cilium also brings the upcoming gateway API (opt-in) as technical preview.

> * The Harbor container registry can now be rollwed out with each cluster.

Reviewer: ```suggestion
* The Harbor container registry can now be rolled out with each cluster.
```

Reviewer 2: Fixed by https://github.com/SovereignCloudStack/release-notes/pull/9/commits/3839913eab87914241c670261d659037cf0ba025Fixed by

* The clusters can use a registry as cache to upstream dockerhub or gcr registries.
* The cluster management now works also on OpenStack clouds with a custom CA.
* Storage snapshots are supported now (fix was also backported to maintained branches).
* Diskless flavors are supported everywhere (cluster-management, health-monitor).

> * etcd defragmenetation and backup.

Reviewer: ```suggestion
* etcd defragmentation and backup.
```

Reviewer 2: Fixed by https://github.com/SovereignCloudStack/release-notes/pull/9/commits/3839913eab87914241c670261d659037cf0ba025Fixed by

* Controls for pod and service IP ranges.

### Preview: Cluster-Stacks
The old scripts that are used to create, change and delete Kubernetes clusters with
cluster-api will be replaced by a proper Operator in the next release.
Design document and technical preview code can be found in the
[cluster-stacks](https://github.com/SovereignCloudStack/cluster-stacks)
and [cluster-stack-operator](https://github.com/SovereignCloudStack/cluster-stack-operator)
repositories. This solution will fit more nicely into the CNCF landscape and
also allow for easier support of IaaS solutions that do not comply to our SCS
IaaS standards.

### Operations and IAM related

* A number of improvements when using identity federation via OIDC has been added, including
  addressing openstack CLI usage with PKCE Device Authz Grant, logout, and the usage of a
  proxy realm in keycloak. Improvements have been contributed to upstream keystone.
* With the `openstack-resource-manager` a new day 2 operations tool has been added.

>   Furthermore a osism role for tuned to optimize system profiles is now present.

Reviewer: ```suggestion
  Furthermore an osism role for tuned to optimize system profiles is now present.
```

Reviewer 2: Fixed this one along with the other typos that you found and some that ispell found in commit https://github.com/SovereignCloudStack/release-notes/pull/9/commits/3839913eab87914241c670261d659037cf0ba025.
I should have done this last night, sorry.

* The [openstack-flavor-manager](https://github.com/osism/openstack-flavor-manager) is now able to create all standard, mandatory SCS flavors for you.
* Scaphandre Prometheus Exporter has been added to export power consumption metrics more easily.
* To optimize system profiles an osism role for tuned is now present.
* Full support for air-gapped installation and operation of environments.

> * A migration script and guide for moving from R4 to R clusters is available.

Reviewer: ```suggestion
* A migration script and guide for moving from R4 to R5 clusters is available.
```

> * Metering has been improved and a reference billing API implementation is available as PoC.

Reviewer: ```suggestion
* Metering has been improved and a reference billing API implementation is available as technical preview.
```

### SCS Developer focused improvements (Cloud-in-a-Box, testbed and k8s cluster management)

* Documentation on testbed and Cloud-in-a-Box have been reworked.

> * Reflecting CiaB's usage as edge cloud solution, it now receives more automated testing.

Reviewer: I've been discussing this with @berendt and the term 'edge cloud appliance' is more fitting than solution, since for a solution various aspects (such as orchestration of multiple CiaBs is missing). 

```suggestion
* Reflecting CiaB's usage as edge cloud appliance, it now receives more automated testing.
```

## Upgrade/Migration notes

> * For the IaaS reference implementation, please refer to the [OSISM 5.0.0 Upgrade Notes](https://release.osism.tech/notes/6.0.0.html#upgrade-notes).

Reviewer 2: ```suggestion
* For the IaaS reference implementation, please refer to the [OSISM 6.0.0 Upgrade Notes](https://release.osism.tech/notes/6.0.0.html#upgrade-notes).
```
(OSISM 6.0.0, not 5.0.0).

and the (previously included) [v1 standard flavors](https://github.com/SovereignCloudStack/standards/blob/main/Standards/scs-0103-v1-standard-flavors.md) -- which includes the [new SSD flavors](https://github.com/SovereignCloudStack/standards/blob/main/Standards/scs-0110-v1-ssd-flavors.md$a), the [v1 entropy standard](https://github.com/SovereignCloudStack/standards/blob/main/Standards/scs-0101-v1-entropy.md). We have also split image naming and standard image recommendations into [v1 standars images]https://github.com/SovereignCloudStack/standards/blob/main/Standards/scs-0104-v1-standard-images.md.

Requirements for [k8s version recency](https://github.com/SovereignCloudStack/standards/blob/main/Standards/scs-0210-v1-k8s-new-version-policy.md), [default storage class](https://github.com/SovereignCloudStack/standards/blob/main/Standards/scs-0211-v1-kaas-default-storage-class.md) as well as requirements to the [container registry](https://github.com/SovereignCloudStack/standards/blob/main/Standards/scs-0212-v1-requirements-for-container-registry.md) have been captured.

> The IAM area has ssen ADRs on the chosen architecture.

Reviewer: ```suggestion
The IAM area has seen ADRs on the chosen architecture.
```

Reviewer 2: Fixed by https://github.com/SovereignCloudStack/release-notes/pull/9/commits/3839913eab87914241c670261d659037cf0ba025
